### ISSUE TITLE
Highlight Color Every Other Line

### ISSUE THREAD

**Issue by DebSec** [Izheil/Firefox-57-full-dark-theme-with-scrollbars#21]
In the about:config and the bookmarks library I have highlighted colors every other line. I'm using the full theme and I'm having issues tracking down what to change to have it with no bright colors. I believe it is in the usercontent.css

https://ibb.co/jGK7o0
https://ibb.co/g3ekaf

**Comment by Izheil** [comment]
I can't seem to replicate the issue, it shows them dark for me.
Are you using any other CSS code apart from this theme, or any other theme that isn't one of firefox personas (lightweight themes), such as australis?
Are you using an OS theme that changes colors to white?

[About:config](https://i.imgur.com/P1Jxrt9.png)
[Catalog](https://i.imgur.com/gR3VfuC.png)

**Comment by DebSec** [comment]
You are correct. I have a plethora of edits. The code did not help. Thanks for taking your time to respond. I upped my chrome file for you to throw in another profile so you can view what I have. I have a great layout and code I'm sure you will be interested in either way.  If you happen to spend time on it and find the issue I would appreciate the feedback on the corrections needed. Password is your username. Izheil
Thanks. 

https://send.firefox.com/download/06bdfebd71/#uVLgdT5dXY_h2eZNOiWRuw

**Comment by DebSec** [comment]
I use the Firefox dark theme and compact mode. I have the title bar disabled and the menu bar enabled. My O/S is Debian. This is also on a new profile. I'm stumped myself. I also use bleachbit to ensure there are no temp files causing issues. Your the first I have shared the files with so far. I have access to other o/s's. When I use ShadowFox theme I do not get this problem. I'm stumped. Thanks for your input and efforts.

**Comment by Izheil** [comment]
Well, you can try adding these lines:

This ones on your colors.css file (userchrome.css in this repository), to change the library colors:
<pre>#placeContent treechildren::-moz-tree-row(odd) {
  background-image: linear-gradient(#333, #333) !important}</pre>

and this one inside the about:config section in usercontent.css (placing them after the odd ones, somewhere around line 666)
<pre>#configTreeBody::-moz-tree-row(even) {
	background-image: linear-gradient(rgba(0,0,0,0.05), rgba(0,0,0,0.05)) !important}</pre>

(Looks like I copied the wrong rule in the other post for the about:config, so use this one instead)

**Comment by DebSec** [comment]
Izheil,
I tried the code as directed to no avail. I also removed any lines that referred to odd or even in those two files with no progress. It seems to be a off white background with white text. I played with colors  again that were lighter swapping them with darker colors with no luck either. In the beginning I was using Developers Toolbox to try to pinpoint the code but it would only let me pick general areas in about:config and the Library. I had to import the usercontent.css  into Dev Toolbox and still had no progress there either. The app would not show changes while live editing the usercontent.css. BTW I'm on Firefox 60.2.2esr (64-bit).
Respectfully DebSec

**Comment by Izheil** [comment]
So you are using Firefox ESR?
This theme was only aimed for regular firefox, although it should still work for all variations of it, so that's probably not the issue here.
In either case, it should be a treechildren selector. When using the toolbox it will select all the items in the list as treechildren, that's normal.

What I don't get is why the odd rows are getting their color changed but not the even ones?
Specially on the library window, the even rows get differently colored than the background color of the list, which shouldn't be happening at all, since they just shouldn't get colored to inherit the background color.

If it was a problem with the selector, the other code should have fixed it, so if it didn't it has to be something external, and as far as I can tell, none of the other .css files you sent had any mention to any of those selectors.
Another question is why is it affecting userchrome related content and usercontent related one. If it was some .css fault, only the library one would be affected, but that doesn't seem like the case.
The only conclusion I can get to is that there is something external from firefox chrome folder changing those things, be it the OS theme (although it would be rare that it would override things that you colored with the code from the other post), or some userstyle from an addon (since ESR can still use XUL based extensions, it could technically modify those things, so make sure you have all userstyles disabled), since they would have higher relevance when being applied.

Either way, worst case scenario you can use shadowfox to change the colors and disable the things you don't like from it.

Since this is more of an issue of a mix of external codes, I'm not sure that I can't help you more, since I can't even replicate the issue, and I can't know what things are actually affecting your firefox for sure, but it certainly doesn't to be related with a problem with this theme.

Hope you have some luck figuring what's causing it.

**Comment by DebSec** [comment]
Izheil,
I tried the chrome folder in Windows 7 Ultimate Firefox V 62 and the problem is not there as you said. I then tried the chrome folder on another Linux laptop and the issues persisted. This must be a code issue between Windows and Linux. I have tried other variations of themes and have not come across this problem before. If I remove your userContent.css and replace it with ShadowFox userContent.css the problem is gone in the about:config page. I noticed **my** chrome files did not look that great and needed work in Windows for the menu background and the base64 image buttons for the max,min,close buttons. If you would keep this open for a bit maybe I can find how to edit your code to remove the line variations that are not compatible in Linux. I'm presuming it is a Linux thing as I said.
Respectfully DebSec

**Comment by Izheil** [comment]
I checked shadowfox, and all they did that could be the reason it works for you is that they give the tree elements the background color and make treechildren transparent.

If anything I guess the tree element could be the cause of the issue, so I made it inherit the background color as well. [Try the new changes on both userchrome and usercontent](https://github.com/Izheil/Firefox-57-full-dark-theme-with-scrollbars/commit/1b3bd20040b3a17279232ad3c1667ebe60c54464) and tell me if the issue persists.
